cocos/core/scene-graph/base-node.ts

```diff
@@ -321,7 +321,10 @@ export class BaseNode extends CCObject implements ISchedulable {
 
     // The PrefabInfo object
     @serializable
-    protected _prefab: any = null;
+    protected _prefab: PrefabInfo|null = null;
+
+    @serializable
+    protected _prefabInstance?: PrefabInstance;
```

Comment:
Why we need to store two prefab props separately?
IMO, prefab is not a general need, so adding as less hack as possible to the node will make the core engine more efficient.

Reply:
> Why we need to store two prefab props separately?
> IMO, prefab is not a general need, so adding as less hack as possible to the node will make the core engine more efficient.

Because the fileId is stored in the prefabInfo, the prefabInstance only exist in the root of a prefab. the prefabInfo can be used to distinguish the `prefab node` and `normal node` more conviniently.